### DISCUSSION TITLE
(use case) orbit-db and git-bug?

### DISCUSSION THREAD

**Discussion by ghost** [git-bug/git-bug#952] [Q&A] [answered]
Hi.

#### Idea
If git-bug is thinking in a decentralized, distributed way, I believe we should have a decentralized, distributed database for the reason: of the "consistency" or "value" we have as an open project or "general purpose we think".

Lately, a project that calls my attention is [orbitdb/orbit-db](https://github.com/orbitdb/orbit-db)... orbit-db uses IPFS as its data storage and IPFS Pubsub to automatically sync databases with peers. So... I think... really... that it would be a good choice to distribute issues to several peers because that would be a decentralized, distributed way - which is in line with the purpose of this project.

In addition, as it is a database(orbit-db), it may be that one or more git-bugs - be a kind of client-server, that is, we distribute issues at the same time as we serve as a communication bridge to other platforms(GitHub, Gitlab, Radicle etc): temporarily or permanently. 

In this sense, there would be a need to have a decentralized and distributed database... because... at the same time we are something like a client-server, something similar to mail, matrix(we are not a central server, we are a distributed and decentralized server, so... something similar to mail, matrix) - it makes sense to also have a decentralized database, distributed to facilitate access to our tool - In addition to the reason that it is in line with our value, purpose I mentioned earlier, initially as an open project.
 
Furthermore, if any of these platforms stop working - and... if we have a decentralized and distributed database, we can redistribute information, data even if these platforms are offline. Then... when they become live again (GitHub, Gitlab, Radicle etc) we can sync changes or updates to those platforms with our database

This facilitates the high availability feature, data replication, interopability that are characteristics of protocols such as IPFS, Hypercore, Matrix - or as my argument revolves around .... of the values, purposes of git-bug as an open project... because "I'm somehow trying to convince... by what I imagine to be interesting, viable."

#### why is this feature interesting?
1. So... databases such as orbit-db were designed for data replication, which facilitates the interoperability resource so present in protocols like: matrix. So... It would be a good use case for things like git-bug to utilize things like orbit-db
2. Git-bug doesn't have a database... So... would like to add this new feature to git-bug with orbit-db
3. I would like to contribute to 2 projects git-bug and orbit-db. So... if you all want I can submit a PR git-bug with orbit-db or so... I can add other protocols perhaps with ongoing apis that are already written in go for hypercore, solid-project, remote-storage - if there is a need for other network protocols. *Please, just confirm if you all this is interesting to be added*
4. I talked about these ideas and got some interesting feedback: [p2p storage in git-bug](https://github.com/MichaelMure/git-bug/discussions/888), [New bridges for different platforms/services](https://github.com/MichaelMure/git-bug/discussions/871), [Is git-bug something federative? something like matrix or email?](https://github.com/MichaelMure/git-bug/discussions/897), I wanted to know some feedback on this new idea

#### question/feedback
- I would like to know what you all think of the idea... *what you all think of the idea?*
- Why doesn't git-bug have a database like orbit-db?

**Comment by MichaelMure** [comment, marked as answer]
AFAIK orbit-db is not that much permanent (aka would lose all the data if there is no peer remaining), which could be a problem for a bugtracker.

However, that would be a really cool experiment. The only thing you need really is an alternative implementation of a Repo (https://github.com/MichaelMure/git-bug/blob/master/repository/repo.go), which could really be anything. You could reuse some existing pieces (RepoStorage, RepoBleve ...) and replace the part that matters (RepoData mainly).

It could be orbit-db, some of the other you mentioned, a classic database, a browser storage ...

**Reply by ghost** [reply]
Hi MichaelMure, thank you for feedback.
